### ISSUE TITLE
Unexpected error message from collection expression cast

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#71490]
**Version Used**: 
https://github.com/dotnet/roslyn/commit/754f1b02295bdfcb8161665229a3530e6d3ba04b

**Steps to Reproduce**:
```C#
using System;

public class C 
{
    private DateTimeOffset _d;
    
    public void M() 
    {
        ReadOnlySpan<DateTime> x = [_d];
        var y = (ReadOnlySpan<DateTime>)[_d];
    }
}
```

**Expected Behavior**:
Both x and y produce the same error that's descriptive about not being able to convert a DateTimeOffset to a DateTime.

**Actual Behavior**:
x produces "error CS0029: Cannot implicitly convert type 'System.DateTimeOffset' to 'System.DateTime'"
y produces "error CS9174: Cannot initialize type 'ReadOnlySpan<DateTime>' with a collection expression because the type is not constructible."
